### PR TITLE
feat(eslint-rules): ngOnChanges require members to use Input bind

### PR DESCRIPTION
**Pull request by m-thompson-code**
### Description

ngOnChanges require members to use Input bind:

1. fix will either add missing Input bind or add a member with Input bind

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ You can learn about how to create your eslint rules by following [this walkthrou
 
 ## Development
 
-When creating eslint rules, you should use [AST Explorer](https://astexplorer.net/) to create your rules.
+When creating eslint rules, you should use [AST Explorer](https://astexplorer.net/) while create your rules.
```
Comment (vdtdev) [inline README.md]: the wording here is a bit wonky

File: tools/eslint-rules/rules/angular/on-changes-use-input-bind.ts [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+          // and value isn't a class member.
+          // Fix will add missing member with Input bind.
+          if (!propertyDefinition) {
+            context.report({
+              node: property,
+              messageId: 'simpleChangeExcludesProperty',
+              fix(fixer) {
+                // const startOfBody = node.body.range[0] + 1;
+                // const firstElement = node.body.body[0].range[0];
+                // // const spacing = context.getSourceCode().getText().substring(startOfBody, firstElement);
```
Comment (vdtdev) [inline tools/eslint-rules/rules/angular/on-changes-use-input-bind.ts]: is this code still needed?

File: tools/eslint-rules/utilities/member-expression/get-member-expressions-from-node.ts [hunk 3]
```diff
@@ -0,0 +1,627 @@
+import { AST_NODE_TYPES, TSESTree } from '@typescript-eslint/utils';
```
Comment (vdtdev) [inline tools/eslint-rules/utilities/member-expression/get-member-expressions-from-node.ts:1]: are all of these overloads actually being used? xD

File: tools/eslint-rules/utilities/member-expression/types.ts [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+ * General function that returns MemberExpressions.
+ * Normally first argument is a `TSESTree` node, but could be empty
+ * if function always returns the same result like an empty array.
+ */
+export type MemberExpressionGetter = (
+  ...args: unknown[]
+) => TSESTree.MemberExpression[];
+
+/**
+ * Maps every `AST_NODE_TYPES` to its `TSESTree` equivalent.
```
Comment (vdtdev) [inline tools/eslint-rules/utilities/member-expression/types.ts]: Can you add to this comment to explain when folks should use TESNodes?

**Review by vdtdev — APPROVED** [review]
LGTM other than some readme verbage and a few general questions

File: tools/eslint-rules/utilities/member-expression/get-member-expressions-from-node.ts (hunk 3 above)
Comment (m-thompson-code) [reply tools/eslint-rules/utilities/member-expression/get-member-expressions-from-node.ts:1]: "yes"

File: tools/eslint-rules/rules/angular/on-changes-use-input-bind.ts (hunk 2 above)
Comment (m-thompson-code) [reply tools/eslint-rules/rules/angular/on-changes-use-input-bind.ts]: nope, removed~

File: tools/eslint-rules/utilities/member-expression/types.ts (hunk 4 above)
Comment (m-thompson-code) [reply tools/eslint-rules/utilities/member-expression/types.ts]: Removed, so neverrr~~

Removed in favor for the built-in `TSESTree.Node` type

**Review by vdtdev — APPROVED** [review]
+1

**Pull request merged**